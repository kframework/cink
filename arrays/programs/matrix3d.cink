int main() {
  int a[2][3][4];
  a[1][2][3] = 7;
  cout << a[1][2][3] << endl << "a[0][1][3] = ";
  cin >> a[0][1][3];
  cout << a[0][1][3] << endl;
  return 0;
}
