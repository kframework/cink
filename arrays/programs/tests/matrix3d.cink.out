7
a[0][1][3] = 8
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
    <store>
        seq-array ( 2 , #symLoc(2) ) |-> array of 2 (array of 3 (array of 
          4 int))
        select ( (seq-array ( 2 , #symLoc(2) )) , 0 ) |-> array of 3 (array of
           4 int)
        select ( (seq-array ( 2 , #symLoc(2) )) , 1 ) |-> array of 3 (array of
           4 int)
        select ( (select ( (seq-array ( 2 , #symLoc(2) )) , 0 )) , 1 ) |-> 
          store ( (const-array ( 4 , 0 )) , 3 , 8 )
        select ( (select ( (seq-array ( 2 , #symLoc(2) )) , 1 )) , 2 ) |-> 
          store ( (const-array ( 4 , 0 )) , 3 , 7 )
        #symScalLoc(1) |-> lambda ( .Decls @ (((((((int (((a [ 2 ]) [ 3 ]) [ 4
           ])) ;) (((((a [ 1 ]) [ 2 ]) [ 3 ]) = 7) ;)) ((((cout << (((a [ 1 ])
           [ 2 ]) [ 3 ])) << endl) << "a[0][1][3] = ") ;)) ((cin >> (((a [ 0 ]
          ) [ 1 ]) [ 3 ])) ;)) (((cout << (((a [ 0 ]) [ 1 ]) [ 3 ])) << endl) 
          ;)) (return 0 ;)) )
    </store>
</T>
